### PR TITLE
API Poll Participation

### PR DESCRIPTION
**Pull request by HankG**
There was no method in the original API spec to do this so this is what I came up with...

### CODE REVIEW

File: spec/integration/api/post_interactions_controller_spec.rb [hunk 1]
```diff
@@ -291,4 +291,91 @@
...
+
+    it "fails with bad post id" do
+      post(
+        api_v1_post_vote_path("999_999_999"),
+        params: {
+          poll_answer_id: @poll_answer.id,
+          access_token:   access_token
+        }
+      )
+      expect(response.status).to eq(422)
```
Comment (frankrousseau) [inline spec/integration/api/post_interactions_controller_spec.rb]: When url is wrong it seems better to send a 404 error.

Comment (HankG) [reply spec/integration/api/post_interactions_controller_spec.rb]: I've corrected this in the new version.

**Comment by HankG** [comment]
I added a little extra handling around the find-post line to catch that "record not found" specifically so now it returns 404 in that instance.  I've also squashed this so it's ready for integration.

**Comment by frankrousseau** [comment]
Great, thank you!

**Pull request merged**